### PR TITLE
Slew to home blocking

### PR DESCRIPTION
**Pull request by wtgee**
Provides a parameter that allows the `slew_to_home` command to block until home is reached. Defaults to `False` for backwards compatibility.

Replaces some custom blocking in various places with new parameter.

Pulled from #859.

### CODE REVIEW

**Comment by jamessynge** [comment]
FWIW, I'm reminded by reading the description of this PR that I encountered a problem where slewing never ended, so the dome was stuck open until I noticed a day+ later.

File: pocs/mount/mount.py [hunk 1]
```diff
@@ -514,28 +511,38 @@ def slew_to_target(self):
 
         return success
 
-    def slew_to_home(self):
+    def slew_to_home(self, blocking=False):
```
Comment (jamessynge) [inline pocs/mount/mount.py]: Should this have a maximum time to block (e.g. 10 minutes)?

Comment (wtgee) [reply pocs/mount/mount.py]: Yes, agreed. I almost marked this WIP until I could add that in as it does seem like a potential bad failure. I'll add before merge and re-request a review. Thanks!

Comment (wtgee) [reply pocs/mount/mount.py]: I've added a timeout, however it's not clear to me what the behavior should be. We slew to home before we park, so if we have a timeout while slewing to home we can't actually then attempt a park.  Note also that #889 potential adds some confusion to this as that new park behaviour is based off movement from the home position.

> FWIW, I'm reminded by reading the description of this PR that I encountered a problem where slewing never ended, so the dome was stuck open until I noticed a day+ later.

IIRC you were stuck in a slew to target but not a slew to home. We would need to add a timeout there as well.

**Pull request merged**